docs/proposals/27-cloudprofile-bastion-section.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Add Optional Bastion Section To CloudProfile
+gep-number: 27
+creation-date: 2024-05-08
+status: implementable
+authors:
+- "@hebelsan"
+- "@kon-angelo"
+- "@AndreasBurger"
+reviewers:
+- "@rfranzke"
+- "@timuthy"
+---
+
+# GEP-27: Add Optional Bastion Section to CloudProfile
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Prerequisites](#prerequisites)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+
+## Summary
+
+Our plan is to extend the gardener core CloudProfile with a new bastion section that will allow us to create a bastion instance with a specific machine image and machine type that will be referenced in that new section.
+
+## Motivation
+
+We are working on updating the bastion images used in our controllers.
+Some of these machine images were hardcoded in the controller.
+We were using different image "families" in some providers like `Debian`, and in others `GardenLinux`.
+We had "custom" logic per provider to determine the "latest" image and so on.
+
+Our plan is to extend the gardener core CloudProfile with a new bastion section that will allow us to create a bastion instance with a specific machine image and machine type that will be referenced in that new section.
+
+There are some advantages to this approach compared to the current one:
+- Consistently use GardenLinux image where possible
+- Easier to maintain
+- Known systemd and ssh requirements for the images
+
+Drawbacks:
+- Bastion is using larger machine types and larger images
+
+The process of looking for the first "supported" version is something we agree with.
+However, without such a place for the machine image, we will depend on the implicit order of the MachineImage array.
+
+### Goals
+
+- Unify bastions across all providers
+- Have a clear definition which image and machine type to use for the bastion
+
+### Prerequisites
+
+Newer GardenLinux versions have ssh disable by default.
+In order for the bastion to work with newer GardenLinux versions we need to update the UserData content that is generated by the [gardenlet bastion controller](https://github.com/gardener/gardener/blob/38143e7b9f4913d6ae11072064bb13a39b2ef17f/pkg/gardenlet/controller/bastion/reconciler.go#L269).
+Adding the following line should be enough:
+```bash
+systemctl start ssh
+```
+
+## Proposal
+
+We propose to add a new bastion section to the CloudProfile which specifies the bastion creation parameters.
+For the machine image, we want include a reference to one of the `machineImages` in the CloudProfile. We can then use the CloudProfile upon request, to deref that image to the provider specific reference.
```

Comment:
Only ref of the image, or also machine type?

Reply:
Refined the proposal section.